kettle-plugins/common/ui/src/main/java/org/pentaho/big/data/plugins/common/ui/MultishimNamedCluster.java

```diff
@@ -0,0 +1,188 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.big.data.plugins.common.ui;
+
+import org.apache.commons.io.FileUtils;
+import org.pentaho.di.core.variables.VariableSpace;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.ui.spoon.Spoon;
+import org.pentaho.hadoop.shim.api.cluster.NamedCluster;
+import org.pentaho.hadoop.shim.api.cluster.NamedClusterService;
+import org.pentaho.metastore.api.IMetaStore;
+import org.pentaho.metastore.api.exceptions.MetaStoreException;
+import org.pentaho.metastore.stores.delegate.DelegatingMetaStore;
+import org.pentaho.metastore.stores.xml.XmlMetaStore;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+public class MultishimNamedCluster {
+
+  public static Class<?> PKG = MultishimNamedCluster.class;
+  public static final String STRING_NAMED_CLUSTERS = BaseMessages.getString( PKG, "NamedClusterDialog.HadoopClusters" );
+
+  private Spoon spoon = Spoon.getInstance();
+  private NamedClusterService namedClusterService;
+  private IMetaStore metaStore;
+  private VariableSpace variableSpace;
+
+  public MultishimNamedCluster( VariableSpace variableSpace, IMetaStore metaStore, NamedClusterService namedClusterService ) {
+    this.namedClusterService = namedClusterService;
+    this.metaStore = metaStore;
+  }
+
+  public String newNamedCluster( String name ) {
+
+    NamedCluster nc = namedClusterService.getClusterTemplate();
+    nc.setName( name );
+    if ( variableSpace != null ) {
+      nc.shareVariablesWith( (VariableSpace) variableSpace );
+    } else {
+      nc.initializeVariablesFrom( null );
+    }
+
+    try {
+      if ( nc.getConfigId() != null ) {
+        delNamedCluster( metaStore, nc );
+      }
+      String newClusterId = generateNewClusterId( null );
+      nc.setConfigId( newClusterId );
+      addConfigProperties( nc );
+      saveNamedCluster( metaStore, nc );
+    } catch ( Exception e ) {
+      /*commonDialogFactory.createErrorDialog( spoon.getShell(),
+          BaseMessages.getString( PKG, SPOON_DIALOG_ERROR_SAVING_NAMED_CLUSTER_TITLE ),
+          BaseMessages.getString( PKG, SPOON_DIALOG_ERROR_SAVING_NAMED_CLUSTER_MESSAGE, nc.getName() ), e );*/
+      spoon.refreshTree();
+      return nc.getName();
+    }
+
+
+    return nc.getName();
+  }
+
+  private void saveNamedCluster( IMetaStore metaStore, NamedCluster namedCluster ) {
+    try {
+      namedClusterService.create( namedCluster, metaStore );
+    } catch ( MetaStoreException e ) {
+      /*commonDialogFactory.createErrorDialog( spoon.getShell(),
```

Comment:
Can these comments be deleted, or are they there for documentation purposes?

Reply:
Peter. I left those comments as a reference for now. That's what the current code based on the SWT UI does. I will know more about what to do with those exceptions as James progresses on the UI front.